CMake/ctest_script.cmake

```diff
@@ -157,6 +158,12 @@ IF( NOT DEFINED N_PROCS )
     ENDIF()
 ENDIF()
 
+# Set the number of processors
+IF( NOT DEFINED N_PROCS_BUILD )
+     IF ( DEFINED $ENV{N_PROCS_BUILD} )
+        SET( N_PROCS_BUILD $ENV{N_PROCS_BUILD} )
+     ENDIF()
```

Comment:
else SET( N_PROCS_BUILD ${N_PROCS} )
Just make sure N_PROCS_BUILD is always defined.

Reply:
We would have to set it for every machine. This is intentionally on the safe side.

Reply:
Not fully understand your intention. I mean if both N_PROCS_BUILD and a corresponding environment variable is not defined, set N_PROCS_BUILD = N_PROCS

Reply:
I think Ye's point is that if his suggested 'else' branch is added here, then N_PROCS_BUILD is defined through all code paths, and and be used directly to simplify the next section that he commented on.